Simple Incremental/Assets/Scripts/Structs/PlayerTarget.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public struct PlayerTarget

```

Comment:
This struct does not seem to add value. 

There is no need to group these 3 objects together. You need to know your current damage, which should be calculated in HelperAttackEnemy, and you need to know the CharacterHealth, which should be returned from your EnemyTargeting script. distance needs to be calculated at the time of the attack anyways and can be done by calling the transforms of the respective components.